tests/contrib/botocore/test.py

```diff
@@ -201,13 +205,40 @@ def test_lambda_client(self):
         assert spans
         span = spans[0]
         self.assertEqual(len(spans), 1)
-        self.assertEqual(span.get_tag('aws.region'), 'us-east-1')
+        self.assertEqual(span.get_tag('aws.region'), 'us-west-2')
         self.assertEqual(span.get_tag('aws.operation'), 'ListFunctions')
         assert_is_measured(span)
         assert_span_http_status_code(span, 200)
         self.assertEqual(span.service, 'test-botocore-tracing.lambda')
         self.assertEqual(span.resource, 'lambda.listfunctions')
 
+    def test_inject_trace_no_context(self):
```

Comment:
:+1: these tests are good for testing the helper, but I'd also like to see tests for an actual lambda invocation with botocore if possible to make sure we aren't breaking or altering the original functionality.

It looks like [moto supports mock lambda invocations](http://docs.getmoto.org/en/latest/index.html?highlight=lambda#currently-implemented-services) so you should be able to just [call lambda normally](https://boto3.amazonaws.com/v1/documentation/api/latest/reference/services/lambda.html) and make sure it doesn't fail 🙂

Reply:
I started that way, but didn't manage to get it working. Got some errors that I couldn't figure out in a day. I needed to create the lambda first and then invoke it but the call to create the lambda failed with an error I couldn't figure out https://app.circleci.com/pipelines/github/DataDog/dd-trace-py/3217/workflows/c8e3965a-08ad-4d44-ad05-a3a10057b676/jobs/366881 in https://github.com/DataDog/dd-trace-py/commit/c3edbef0794ebc7e9a0afedaf8a4d6cea04a8bf7

I can give it another try after addressing you implementation concerns

Reply:
ah gotcha! Yeah I'm not very familiar with boto/moto myself so I have no idea how difficult it is to mock out a lambda. It would be really great to have this kind of test though 👍

Reply:
@Czechh did you try giving this another go? I'd really like to have it if possible but I think it's okay if it's too much hassle to get it working. If it is too difficult would you be able to put together a manual test case to demonstrate it working? Having it in https://github.com/DataDog/trace-examples or in a gist would be great.